app/services/open_graph.rb

```diff
@@ -0,0 +1,77 @@
+# This class provides methods for getting structured metadata properties for
+# Open Graph usage
+#
+# @note A wrapper around the metainspector gem
+class OpenGraph
+  delegate :meta, :meta_tags, :images, to: :page
+
+  attr_accessor :page, :tags
+
+  DEFAULT_METHODS = %i(author title description).freeze
+
+  def initialize(url)
+    @page = MetaInspector.new(url)
+    @tags = meta_tags
+  end
+
+  # the optional "use_best" argument is used to determine if we should use the
+  # "best" option availabile for said method this works by picking the most
+  # available (first?) image, description, or author (respectively), if defined
+  # with the og:* or twitter:* metatags. example page.title(true) would look in
+  # said areas the default only looks at the head or default metatag.
+
+  DEFAULT_METHODS.each do |method_name|
+    define_method(method_name) do |use_best = false|
+      if use_best
+        page.public_send("best_#{method_name}".to_sym)
+      else
+        page.public_send(method_name)
+      end
+    end
+  end
+
+  def meta_for(data)
+    page.meta[data]
+  end
+
+  def favicon
+    images.favicon
+  end
+
+  def properties
+    tags ||= meta_tags
```

Comment:
This should probably be `@tags ||= meta_tags` or `self.tags ||= meta_tags`, because as written this creates and sets a local variable.

```ruby
class Hello
  attr_accessor :tags
  def initialize
    tags ||= [1,2,3]
  end
end

puts Hello.new.tags.inspect
=> nil
```

Reply:
ah -- good call